### DISCUSSION TITLE
MauiImage resource as stream

### DISCUSSION THREAD

**Discussion by AHComp** [dotnet/maui#11645] [Q&A]
I need access to resources from type MauiImage. If i insert the resource as embedded resource it works see here.

![image](https://user-images.githubusercontent.com/35998144/203938984-d45e7373-532c-4b02-8866-9bb9cb415685.png)

Now i have the resource double in the project as MauiImage and as embedded resource.

How i can access the MauiImage resource to get a stream to create a IImage or direct an IImage?

**Comment by karthikraja-arumugam** [comment]
Hi @AHComp , Have you found any solution? I'm also looking for this.

**Reply by AHComp** [reply]
No I opened a issue but this was immediately closed by the dev team.

**Comment by xtuzy** [comment]
- On Android
MauiImage store as drawable, so you can google how to get them:
![image](https://user-images.githubusercontent.com/17793881/211981353-13f579e0-9848-4d87-9b5a-1c26b64c8d53.png)
Maybe these code can help you:
     - Get id of drawable :https://github.com/dotnet/maui/blob/35f3e66e1ce3e070830d8f1abbda222a1759c215/src/Core/src/Platform/Android/ContextExtensions.cs#L281
     - Then read Bitmap:https://stackoverflow.com/questions/3035692/how-to-convert-a-drawable-to-a-bitmap
     - Maui.Graphics.IImage can be create form this bitmap:https://github.com/dotnet/maui/blob/35f3e66e1ce3e070830d8f1abbda222a1759c215/src/Graphics/src/Graphics/Platforms/Android/PlatformImage.cs#L13

- On Windows, 
i don't know MauiImage store where, but i search this file, you can try these method in it:
https://github.com/dotnet/maui/blob/main/src/Core/src/ImageSources/FileImageSourceService/FileImageSourceService.Windows.cs
- On iOS
It maybe as Assets, so see [xamarin.ios doc](https://learn.microsoft.com/en-us/xamarin/ios/app-fundamentals/images-icons/displaying-an-image?tabs=macos) 
![image](https://user-images.githubusercontent.com/17793881/211982325-58170cbe-b267-436e-a614-6b7081425752.png)
    - some code https://github.com/dotnet/maui/blob/35f3e66e1ce3e070830d8f1abbda222a1759c215/src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs#L39
